ethereumj-core/src/main/java/org/ethereum/sync/SyncManager.java

```diff
@@ -269,10 +316,61 @@ private void produceQueue() {
             } catch (InterruptedException e) {
                 break;
             } catch (Throwable e) {
-                logger.error("Error processing block {}: ", wrapper.getBlock().getShortDescr(), e);
-                logger.error("Block dump: {}", Hex.toHexString(wrapper.getBlock().getEncoded()));
+                if (wrapper != null) {
+                    logger.error("Error processing block {}: ", wrapper.getBlock().getShortDescr(), e);
+                    logger.error("Block dump: {}", Hex.toHexString(wrapper.getBlock().getEncoded()));
+                } else {
+                    logger.error("Error processing unknown block", e);
+                }
+            }
+        }
+    }
+
+    /**
+     * Processing new blocks from queue
+     */
+    private void newBlocksDistributeLoop() {
```

Comment:
M.b. let's move blocks and tx broadcasting logic from `SyncManager` to let's say `ChannelManager`. 
Those functions has actually few in common with syncing and more with networking